### PR TITLE
Fix several issues discovered with analyze tools

### PR DESCRIPTION
**Pull request by luis-pereira**

### CODE REVIEW

**Comment by luis-pereira** [comment]
Analyze & debug in progress

File: qtxdg/xdgdesktopfile.cpp [hunk 1]
```diff
@@ -1480,8 +1480,16 @@ void XdgDesktopFileCache::initialize(const QString& dirName)
 XdgDesktopFile* XdgDesktopFileCache::load(const QString& fileName)
 {
     XdgDesktopFile* desktopFile = new XdgDesktopFile();
-    desktopFile->load(fileName);
-    return desktopFile;
+
+    Q_CHECK_PTR(desktopFile);
+    if (desktopFile && desktopFile->load(fileName))
+    {
+        desktopFile->load(fileName);
```
Comment (paulolieuthier) [inline qtxdg/xdgdesktopfile.cpp]: Twice?

Comment (luis-pereira) [inline qtxdg/xdgdesktopfile.cpp]: ooooppps, didn't commit. It's time to rest :wink:

File: qtxdg/xdgdesktopfile.cpp [hunk 2]
```diff
@@ -836,9 +836,11 @@ bool XdgDesktopFile::startDetached(const QStringList& urls) const
     {
     case ApplicationType:
         return d->startApplicationDetached(this, urls);
+        break;
```
Comment (jubalh) [inline qtxdg/xdgdesktopfile.cpp]: if one makes `return` the `break` doesnt have much effect, right?

Comment (luis-pereira) [inline qtxdg/xdgdesktopfile.cpp]: t@jubalh Sure. But the return might be replaced by something else and adding the `break` might be forgot.
Also, it confuses analyzers.

**Comment by luis-pereira** [comment]
It has been working without glitches. It needs to go mainstream.

**Pull request merged**